src/dependency-manager/src/utils/files.ts

```diff
@@ -0,0 +1,37 @@
+import fs from 'fs';
+import path from 'path';
+import untildify from 'untildify';
+
+export const insertFileDataFromRefs = (file_contents: string, file_path: string, config_path: string) => {
```

Comment:
Any reason we need to support yml and json? I'd rather this run in the buildFromJson methods. Stringify the json, regex replace, and parse.

Reply:
I had considered that at one point, but that would cause validation error line numbers to be off for yaml files. I could've looked into some kind of normalizing for that, but it seemed worse than just doing the replace for the actual file type, json or yaml

Reply:
Hmm this took me a sec. I would have thought the opposite. Just to confirm I would expect the following:

```yaml
parameters:
  config: file:./config.json
interfaces:
  invalid: ${{ bug }}
```

Validation error line 4

If we inline multi-line file contents won't this lead to a higher line number?